packages/api/db/tables.sql

```diff
@@ -0,0 +1,28 @@
+-- User tags are associated to a user for the purpose of granting/restricting them
+-- in the application.
+-- Needed per https://www.postgresql.org/message-id/20141203213139.GA9855%40fetter.org
+CREATE TYPE user_tag_type AS ENUM
+(
+  'HasAccountRestriction',
+  'HasPsaAccess',
+  'HasSuperHotAccess',
+  'StorageLimitBytes'
+);
+
+-- A transaction in nftstorage.link perma cache.
+CREATE TABLE IF NOT EXISTS public.perma_cache
```

Comment:
Personally I'd still have two tables, one for tracking the items the user has currently cached and one for tracking the history of what happened. With just one table we lose the ability to accurately track a single item that is cached and deleted and re-cached again.

Reply:
Since this is a paid for service we need to be able to account for everything accurately.

Reply:
We can keep track of everything in here, and with SQL I think it will only add complexity to have both tables.

The first KV we had (current state of things), is essentially everything this table where `deleted_at` is null. The second KV (transaction history) is the table as a whole.

If a user does the flow:
- Post URL_A
- Delete URL_A
- Post URL_A
- Delete URL_A
- Post URL_A

We will end up with 5 rows in the table keeping the entire transaction history for billing period: `inserted_at` -> `deleted_at` || NOW() * size

We will keep single entry with `deleted_at` as null.

The relevant SQL to make this work:

```sql
CREATE UNIQUE INDEX IF NOT EXISTS perma_cache_is_deleted_idx ON perma_cache (user_id, source_url, deleted_at)
WHERE deleted_at IS NOT NULL;
CREATE UNIQUE INDEX IF NOT EXISTS perma_cache_is_not_deleted_idx ON perma_cache (user_id, source_url)
WHERE deleted_at IS NULL;
```